src/Microsoft.TestPlatform.CrossPlatEngine/AttachmentsProcessing/DataCollectorAttachmentsProcessorsFactory.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.VisualStudio.TestPlatform.Common.DataCollector;
+using Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework;
+using Microsoft.VisualStudio.TestPlatform.Common.Logging;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Engine;
+using Microsoft.VisualStudio.TestPlatform.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.TestRunAttachmentsProcessing
+{
+    internal class DataCollectorAttachmentsProcessorsFactory : IDataCollectorAttachmentsProcessorsFactory
+    {
+        private static Uri CoverageUri = new Uri("datacollector://microsoft/CodeCoverage/2.0");
+        private const string CoverageFriendlyName = "Code Coverage";
+
+        public IReadOnlyDictionary<string, IConfigurableDataCollectorAttachmentProcessor> Create(InvokedDataCollector[] invokedDataCollectors)
+        {
+            IDictionary<string, Tuple<string, IConfigurableDataCollectorAttachmentProcessor>> datacollectorsAttachmentsProcessors = new Dictionary<string, Tuple<string, IConfigurableDataCollectorAttachmentProcessor>>();
+            bool addCodeCoverageAttachmentProcessors = true;
+
+            if (invokedDataCollectors != null)
+            {
+                // We order files by filename descending so in case of the same collector from the same nuget but with different versions, we'll run the newer version.
```

Comment:
> We order files by filename descending so in case of the same collector from the same nuget but with different versions, we'll run the newer version. [](http://example.com/codeflow?start=18&length=149)

When would this normally happen? (where we have different version of coverlet referenced)
Is this across projects in the solution?

Reply:
Yep if for instance in VS we have more than one project with different version(we're using the coverage collector as use case)  the idea is to use the up to date one.

Reply:
Would be nice if we could add that detail to the comment as well.